pkgs/applications/editors/pulsar/default.nix

```
@@ -0,0 +1,177 @@
+{ lib
+, stdenv
+, git
+, runtimeShell
+, fetchzip
+, wrapGAppsHook
+, glib
+, gtk3
+, atomEnv
+, xorg
+, libxkbcommon
+, hunspell
+, hunspellDicts
+, useHunspell ? true
+, languages ? [ "en_US" ]
+, nodePackages # We need ASAR to unpack the app.asar file and patch paths
+, python3 # Cursed patch to allow asar unpacking, as the builtin in the unpacked asar isn't there
+, makeDesktopItem
+}:
+
+let
+  owner = "pulsar-edit";
+  pname = "Pulsar";
+  version = "1.103.0";
+
+  sources = {
+    x86_64-linux = fetchzip {
+      url = "https://github.com/${owner}/${pname}/releases/download/v1.103.0/Linux.${pname}-${version}.tar.gz";
+      hash = "sha256-O+mekV2h3mxRPYpIrpoPHQyRDuXgl+En8n8u2yBG8TQ=";
+    };
+    aarch64-linux = fetchzip {
+      url = "https://github.com/pulsar-edit/pulsar/releases/download/v1.103.0/ARM.Linux.pulsar-1.103.0-arm64.tar.gz";
+      hash = "sha256-gajIHUJrjCogcJjJVSKz85/x9inuS9rEMYlLqcolUqg=";
+    };
+  };
```

I would prolly cut this down to just what differs between platforms.
```suggestion
  pname = "pulsar";
  version = "1.103.0";

  sourceDetails = {
    x86_64-linux = {
      tarname = "Linux.${pname}-${version}";
      hash = "sha256-O+mekV2h3mxRPYpIrpoPHQyRDuXgl+En8n8u2yBG8TQ=";
    };
    aarch64-linux = {
      tarname = "ARM.Linux.${pname}-${version}-arm64";
      hash = "sha256-gajIHUJrjCogcJjJVSKz85/x9inuS9rEMYlLqcolUqg=";
    };
  }.${stdenv.hostPlatform.system} or (throw "Don't know how to fetch pulsar src for ${stdenv.hostPlatform.system}!");
```

(the fallback `throw` there should fix the Darwin eval I think)